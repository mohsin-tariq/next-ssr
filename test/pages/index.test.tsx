import React from 'react'
import { render } from '../testUtils'
import mockResponse from '../__mocks__/mockResponse.json'
import emptyMissionIdMockResponse from '../__mocks__/mockResponseEmptyMissionId.json'

import App from '../../pages/index'

import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'



describe('Page render snapshots', () => {
  const mockStore = configureStore()
  let store

  it('Renders page with values"', () => {
    const initialState = {
      filter: {
          land_success: 'true',
          launch_success: 'true',
          launch_year: '2004'
      }
    }
    store = mockStore(initialState)
    const { asFragment } = render(<Provider store={store}><App  data={mockResponse}/></Provider>)
    expect(asFragment()).toMatchSnapshot()
  })
  it('Renders page without values"', () => {
    const initialState = {
      filter: {
          land_success: 'false',
          launch_success: 'true',
          launch_year: '2004'
      }
    }
    store = mockStore(initialState)
    const { asFragment } = render(<Provider store={store}><App/></Provider>)
    expect(asFragment()).toMatchSnapshot()
  })

  
  it('Renders page without missionId without Image"', () => {
    const initialState = {
      filter: {
          land_success: '',
          launch_success: '',
          launch_year: ''
      }
    }
    store = mockStore(initialState)
    const { asFragment } = render(<Provider store={store}><App data={emptyMissionIdMockResponse}/></Provider>)
    expect(asFragment()).toMatchSnapshot()
  })
})