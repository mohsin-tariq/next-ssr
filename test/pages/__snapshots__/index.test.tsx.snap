// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page render snapshots Renders page with values" 1`] = `
<DocumentFragment>
  <div>
    <main>
      <a
        class="heading1"
        href="/"
      >
         SpaceX Launch Program
      </a>
      <div
        class="container"
      >
        <div
          class="left-panel"
        >
          <h3
            class="heading3"
          >
             Filter
          </h3>
          <div
            class="left-panel-item"
          >
            <div
              class="title-divider"
            >
              <p
                class=""
              >
                 Launch Year
              </p>
            </div>
            <div
              class="button-container"
            >
              <a
                class="button "
                href="/filter?&launch_success=true&land_success=true&launch_year=2006"
              >
                2006
              </a>
              <a
                class="button "
                href="/filter?&launch_success=true&land_success=true&launch_year=2007"
              >
                2007
              </a>
              <a
                class="button "
                href="/filter?&launch_success=true&land_success=true&launch_year=2008"
              >
                2008
              </a>
              <a
                class="button "
                href="/filter?&launch_success=true&land_success=true&launch_year=2009"
              >
                2009
              </a>
              <a
                class="button "
                href="/filter?&launch_success=true&land_success=true&launch_year=2010"
              >
                2010
              </a>
              <a
                class="button "
                href="/filter?&launch_success=true&land_success=true&launch_year=2011"
              >
                2011
              </a>
              <a
                class="button "
                href="/filter?&launch_success=true&land_success=true&launch_year=2012"
              >
                2012
              </a>
              <a
                class="button "
                href="/filter?&launch_success=true&land_success=true&launch_year=2013"
              >
                2013
              </a>
              <a
                class="button "
                href="/filter?&launch_success=true&land_success=true&launch_year=2014"
              >
                2014
              </a>
              <a
                class="button "
                href="/filter?&launch_success=true&land_success=true&launch_year=2015"
              >
                2015
              </a>
              <a
                class="button "
                href="/filter?&launch_success=true&land_success=true&launch_year=2016"
              >
                2016
              </a>
              <a
                class="button "
                href="/filter?&launch_success=true&land_success=true&launch_year=2017"
              >
                2017
              </a>
              <a
                class="button "
                href="/filter?&launch_success=true&land_success=true&launch_year=2018"
              >
                2018
              </a>
              <a
                class="button "
                href="/filter?&launch_success=true&land_success=true&launch_year=2019"
              >
                2019
              </a>
              <a
                class="button "
                href="/filter?&launch_success=true&land_success=true&launch_year=2020"
              >
                2020
              </a>
            </div>
          </div>
          <div
            class="left-panel-item"
          >
            <div
              class="title-divider"
            >
              <p
                class=""
              >
                 Successful Launch
              </p>
            </div>
            <div
              class="button-container"
            >
              <a
                class="button active"
                href="/filter?&launch_success=true&land_success=true&launch_year=2004"
              >
                true
              </a>
              <a
                class="button "
                href="/filter?&launch_success=false&land_success=true&launch_year=2004"
              >
                false
              </a>
            </div>
          </div>
          <div
            class="left-panel-item"
          >
            <div
              class="title-divider"
            >
              <p
                class=""
              >
                 Successful Landing
              </p>
            </div>
            <div
              class="button-container"
            >
              <a
                class="button active"
                href="/filter?&launch_success=true&land_success=true&launch_year=2004"
              >
                true
              </a>
              <a
                class="button "
                href="/filter?&launch_success=true&land_success=false&launch_year=2004"
              >
                false
              </a>
            </div>
          </div>
        </div>
        <div
          class="d-flex"
        >
          <div
            class="card-list"
          >
            <div
              class="card"
            >
              <div
                class="image-container"
              >
                <div
                  style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 180px; height: 180px;"
                >
                  <img
                    alt="FalconSat"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                </div>
              </div>
              <div
                class="card-container"
              >
                <p
                  class="title bold"
                >
                   FalconSat #1
                </p>
                <p
                  class="bold"
                >
                   Mission Ids 
                </p>
                <ul>
                  <li>
                    Sector_1
                  </li>
                  <li>
                    Sector_2
                  </li>
                </ul>
                <p
                  class=""
                >
                   
                  <span
                    class="bold"
                  >
                    Launch Year:
                  </span>
                  2006
                </p>
                <p
                  class=""
                >
                   
                  <span
                    class="bold"
                  >
                    Successful Launch:
                  </span>
                </p>
                <p
                  class=""
                >
                   
                  <span
                    class="bold"
                  >
                    Successful landing:
                  </span>
                  true
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer>
      <div
        class="footer"
      >
        <span>
           Developed By: 
          <a
            href="https://www.linkedin.com/in/mohsin-tariq--reactjs/"
            rel="noopener noreferrer"
            target="_blank"
          >
             Mohsin Tariq 
          </a>
        </span>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`Page render snapshots Renders page without missionId without Image" 1`] = `
<DocumentFragment>
  <div>
    <main>
      <a
        class="heading1"
        href="/"
      >
         SpaceX Launch Program
      </a>
      <div
        class="container"
      >
        <div
          class="left-panel"
        >
          <h3
            class="heading3"
          >
             Filter
          </h3>
          <div
            class="left-panel-item"
          >
            <div
              class="title-divider"
            >
              <p
                class=""
              >
                 Launch Year
              </p>
            </div>
            <div
              class="button-container"
            >
              <a
                class="button "
                href="/filter?&launch_year=2006"
              >
                2006
              </a>
              <a
                class="button "
                href="/filter?&launch_year=2007"
              >
                2007
              </a>
              <a
                class="button "
                href="/filter?&launch_year=2008"
              >
                2008
              </a>
              <a
                class="button "
                href="/filter?&launch_year=2009"
              >
                2009
              </a>
              <a
                class="button "
                href="/filter?&launch_year=2010"
              >
                2010
              </a>
              <a
                class="button "
                href="/filter?&launch_year=2011"
              >
                2011
              </a>
              <a
                class="button "
                href="/filter?&launch_year=2012"
              >
                2012
              </a>
              <a
                class="button "
                href="/filter?&launch_year=2013"
              >
                2013
              </a>
              <a
                class="button "
                href="/filter?&launch_year=2014"
              >
                2014
              </a>
              <a
                class="button "
                href="/filter?&launch_year=2015"
              >
                2015
              </a>
              <a
                class="button "
                href="/filter?&launch_year=2016"
              >
                2016
              </a>
              <a
                class="button "
                href="/filter?&launch_year=2017"
              >
                2017
              </a>
              <a
                class="button "
                href="/filter?&launch_year=2018"
              >
                2018
              </a>
              <a
                class="button "
                href="/filter?&launch_year=2019"
              >
                2019
              </a>
              <a
                class="button "
                href="/filter?&launch_year=2020"
              >
                2020
              </a>
            </div>
          </div>
          <div
            class="left-panel-item"
          >
            <div
              class="title-divider"
            >
              <p
                class=""
              >
                 Successful Launch
              </p>
            </div>
            <div
              class="button-container"
            >
              <a
                class="button "
                href="/filter?&launch_success=true"
              >
                true
              </a>
              <a
                class="button "
                href="/filter?&launch_success=false"
              >
                false
              </a>
            </div>
          </div>
          <div
            class="left-panel-item"
          >
            <div
              class="title-divider"
            >
              <p
                class=""
              >
                 Successful Landing
              </p>
            </div>
            <div
              class="button-container"
            >
              <a
                class="button "
                href="/filter?&land_success=true"
              >
                true
              </a>
              <a
                class="button "
                href="/filter?&land_success=false"
              >
                false
              </a>
            </div>
          </div>
        </div>
        <div
          class="d-flex"
        >
          <div
            class="card-list"
          >
            <div
              class="card"
            >
              <div
                class="image-container"
              >
                <div
                  style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 180px; height: 180px;"
                >
                  <img
                    alt="FalconSat"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                </div>
              </div>
              <div
                class="card-container"
              >
                <p
                  class="title bold"
                >
                   FalconSat #1
                </p>
                <p
                  class="bold"
                >
                   Mission Ids 
                </p>
                <ul />
                <p
                  class=""
                >
                   
                  <span
                    class="bold"
                  >
                    Launch Year:
                  </span>
                  2006
                </p>
                <p
                  class=""
                >
                   
                  <span
                    class="bold"
                  >
                    Successful Launch:
                  </span>
                </p>
                <p
                  class=""
                >
                   
                  <span
                    class="bold"
                  >
                    Successful landing:
                  </span>
                  true
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer>
      <div
        class="footer"
      >
        <span>
           Developed By: 
          <a
            href="https://www.linkedin.com/in/mohsin-tariq--reactjs/"
            rel="noopener noreferrer"
            target="_blank"
          >
             Mohsin Tariq 
          </a>
        </span>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`Page render snapshots Renders page without values" 1`] = `
<DocumentFragment>
  <div>
    <main />
    <footer>
      <div
        class="footer"
      >
        <span>
           Developed By: 
          <a
            href="https://www.linkedin.com/in/mohsin-tariq--reactjs/"
            rel="noopener noreferrer"
            target="_blank"
          >
             Mohsin Tariq 
          </a>
        </span>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
